File: hashing/find_numbers.js
/**
 * @param {number[]} nums
 * @returns {number[]}
 */
const findNumbers = (nums) => {
    let ans = [];
    let numsSet = new Set(nums);

    for (const num of nums)
    {
        if (!numsSet.has(num + 1) && !numsSet.has(num - 1))
        {
            ans.push(num);
        }
    }

    return ans;
}

File: hashing/find_numbers.java
public int[] findNumbers(int[] nums)
{
    List<Integer> ans     = new ArrayList<>();
    Set<Integer>  numsSet = new HashSet<>();

    for (int num : nums)
    {
        numsSet.add(num);
    }

    for (int num : nums)
    {
        if (!numsSet.contains(num + 1) && !numsSet.contains(num - 1))
        {
            ans.add(num);
        }
    }

    return ans;
}